cli/stacktrace.cpp

```diff
@@ -43,60 +51,92 @@ void print_stacktrace(FILE* output, int start_idx, bool demangling, int maxdepth
+        // 0   test-signalhandler                  0x0000000100dbf42c _Z16print_stacktraceP7__sFILEibib + 124
+
+        /*
+         * examples:
+         * ./test-signalhandler(_Z16print_stacktraceP8_IO_FILEibib+0xa1) [0x55cb65e41464]
+         * ./test-signalhandler(+0xf9d9) [0x55cb65e3c9d9]
+         */
+
         // skip all leading libc symbols so the first symbol is our code
         if (omit_above_own && !own_code) {
             if (strstr(symbolString, "/libc.so.6") != nullptr)
                 continue;
             own_code = true;
-            offset = i; // make sure the numbering is continous if we omit frames
+            offset = i; // make sure the numbering is continuous if we omit frames
         }
-        char * realnameString = nullptr;
-        const char * const firstBracketName     = strchr(symbolString, '(');
-        const char * const firstBracketAddress  = strchr(symbolString, '[');
-        const char * const secondBracketAddress = strchr(firstBracketAddress, ']');
-        const char * const beginAddress         = firstBracketAddress+3;
-        const int addressLen = int(secondBracketAddress-beginAddress);
-        const int padLen     = (ADDRESSDISPLAYLENGTH-addressLen);
-        if (demangling && firstBracketName) {
-            const char * const plus = strchr(firstBracketName, '+');
-            if (plus && (plus>(firstBracketName+1))) {
-                char input_buffer[1024]= {0};
-                strncpy(input_buffer, firstBracketName+1, plus-firstBracketName-1);
-                size_t length = getArrayLength(demangle_buffer);
-                int status=0;
-                // We're violating the specification - passing stack address instead of malloc'ed heap.
-                // Benefit is that no further heap is required, while there is sufficient stack...
-                realnameString = abi::__cxa_demangle(input_buffer, demangle_buffer, &length, &status); // non-NULL on success
+        const char * realnameString = nullptr;
+        if (demangling) {
+            const char * const firstBracketName = strchr(symbolString, '(');
+            if (firstBracketName) {
+                const char * const plus = strchr(firstBracketName, '+');
+                if (plus && (plus>(firstBracketName+1))) {
+                    char input_buffer[1024]= {0};
+                    strncpy(input_buffer, firstBracketName+1, plus-firstBracketName-1);
+                    size_t length = getArrayLength(demangle_buffer);
+                    int status=0;
+                    // We're violating the specification - passing stack address instead of malloc'ed heap.
+                    // Benefit is that no further heap is required, while there is sufficient stack...
+                    realnameString = abi::__cxa_demangle(input_buffer, demangle_buffer, &length, &status); // non-NULL on success
+                }
             }
         }
+
+        const char * const firstBracketAddress = strchr(symbolString, '[');
```

Comment:
an idea is to move this above the realnameString to avoid some string searches and copies etc.. but I don't have a strong opinion.

Reply:
Yes, we could probably exit earlier. But since this is very cold code (it will only ever be run exactly once) and only when the application crashes it should not matter.